src/openms/source/ANALYSIS/ID/SimpleSearchEngineAlgorithm.cpp

```diff
@@ -245,39 +260,73 @@ void SimpleSearchEngineAlgorithm::postProcessHits_(const PeakMap& exp,
       annotated_hits.shrink_to_fit();
     }
 
-#pragma omp parallel for default(none) shared(annotated_hits, exp, fixed_modifications, variable_modifications, peptide_ids, max_variable_mods_per_peptide)
+    bool annotation_precursor_error_ppm = std::find(annotate_psm_.begin(), annotate_psm_.end(), "precursor_error_ppm") !=  annotate_psm_.end();
+    bool annotation_fragment_error_ppm = std::find(annotate_psm_.begin(), annotate_psm_.end(), "median_fragment_error_ppm") != annotate_psm_.end();
+
+#pragma omp parallel for default(none) shared(annotated_hits, exp, fixed_modifications, variable_modifications, peptide_ids, max_variable_mods_per_peptide, annotation_precursor_error_ppm, annotation_fragment_error_ppm)
     for (SignedSize scan_index = 0; scan_index < (SignedSize)annotated_hits.size(); ++scan_index)
     {
       if (!annotated_hits[scan_index].empty())
       {
+        const MSSpectrum& spec = exp[scan_index];
```

Comment:
maybe add a bounds/size check at least before the loop?

Reply:
If the spectrum has been annotated it must exist.